plugins/modules/system/sudoers.py

```diff
@@ -0,0 +1,190 @@
+  sudoers:
+    name: alice-service
+    user: alice
+    commands:
+      - /bin/systemctl restart my-service
+      - /bin/systemctl reload my-service
+    nopassword: false
+
+- name: Revoke the previous sudo grants given to the alice user
+  sudoers:
+      name: alice-service
+      state: absent
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+
+
+class Sudoers(object):
+
+    def __init__(self, module):
+        self.check_mode = module.check_mode
+        self.name = module.params['name']
+        self.user = module.params['user']
+        self.group = module.params['group']
+        self.state = module.params['state']
+        self.nopassword = module.params['nopassword']
+        self.sudoers_path = module.params['sudoers_path']
+        self.file = os.path.join(self.sudoers_path, self.name)
+        self.commands = module.params['commands']
+
+    def write(self):
+        with open(self.file, 'w') as f:
+            f.write(self.content())
+
+    def delete(self):
+        os.remove(self.file)
+
+    def exists(self):
+        return os.path.exists(self.file)
+
+    def matches(self):
+        with open(self.file, 'r') as f:
+            return f.read() == self.content()
+
+    def content(self):
+        if self.user:
+            owner = self.user
+        elif self.group:
+            owner = '%{group}'.format(group=self.group)
+
+        commands_str = ', '.join(self.commands)
+        nopasswd_str = 'NOPASSWD:' if self.nopassword else ''
+        return "{owner} ALL={nopasswd} {commands}\n".format(owner=owner, nopasswd=nopasswd_str, commands=commands_str)
+
+    def run(self):
+        if self.state == 'absent' and self.exists():
+            if not self.check_mode:
+                self.delete()
+            return True
+
+        if self.exists() and self.matches():
+            return False
+
+        if not self.check_mode:
+            self.write()
+
+        return True
+
+
+def main():
+    argument_spec = {
+        'commands': {
+            'type': 'list',
+            'elements': 'str',
+            'default': [],
+        },
+        'group': {},
+        'name': {
+            'required': True,
+        },
+        'nopassword': {
+            'type': 'bool',
+            'default': True,
+        },
+        'sudoers_path': {
+            'type': 'str',
+            'default': '/etc/sudoers.d',
+        },
+        'state': {
+            'default': 'present',
+            'choices': ['present', 'absent'],
+        },
+        'user': {},
+    }
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[['user', 'group']],
+        supports_check_mode=True,
+        required_if=[('state', 'present', ['commands'])],
+    )
+
+    sudoers = Sudoers(module)
+
+    try:
+        changed = sudoers.run()
+        module.exit_json(changed=changed)
+    except Exception as e:
+        module.fail_json(msg=str(e))
```

Comment:
```suggestion
        module.fail_json(msg=to_native(e))
```
This should be accompanied by the following import as well:
```python
from ansible.module_utils.common.text.converters import to_native
```